lib/stm32/g0/adc.c

```diff
@@ -86,8 +86,8 @@ void adc_set_channel_sample_time_selection(uint32_t adc, uint8_t channel, uint8_
 	uint32_t reg32;
 
 	reg32 = ADC_SMPR1(adc);
-	reg32 &= ~(ADC_SMPR_SMPSEL_CHANNEL_MASK << ADC_SMPR_SMPSEL_CHANNEL_SHIFT(channel));
-	reg32 |= (selection << ADC_SMPR_SMPSEL_CHANNEL_SHIFT(channel));
+	if (selection) reg32 |= ADC_SMPR_SMPSEL(channel);
+	else reg32 &= ~ADC_SMPR_SMPSEL(channel);
```

Comment:
since the code follows the https://www.kernel.org/doc/html/v4.10/process/coding-style.html I anticipate that this would need to be converted to `if (...) {\n` form.

See also: https://github.com/libopencm3/libopencm3/blob/master/HACKING#L34-L37

I think @karlp was also wanting to use the `<< _SHIFT` to a macro doing it internally.

Reply:
Done.

> I think @karlp was also wanting to use the << _SHIFT to a macro doing it internally.

The same as above - one-bit fields are shifted - Karl's own words (see above).